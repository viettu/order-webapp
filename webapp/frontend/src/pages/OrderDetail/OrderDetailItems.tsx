import { Box, Text, VStack, Flex, Spacer, Center, StackDivider, Image } from "@chakra-ui/react";
import { IOrderItem } from '../../data';

type OrderDetailItemsProps = {
    items: Array<IOrderItem>
}

const OrderDetailItems: React.FC<OrderDetailItemsProps> = ({items}) => {
    if (!items || items.length === 0) {
        return null
    }

    const getTotalAmount = () => {
        return items.reduce((acc, item) => acc + (item.quantity * item.product.price), 0);
    }

    return (
        <VStack divider={<StackDivider borderColor='gray.200' />}>
            {items.map((item: IOrderItem) => (
                <Flex key={item.id} w={'100%'} direction={'row'} alignItems={'center'}>
                    <Image w={100} h={100}
                        src={`/images/${item.product.image}.png`}
                        objectFit="cover"
                        alt={item.product.title}>
                    </Image>
                    <Flex flex={1} flexWrap={'wrap'} direction={{ base: "column", md: "row" }} ml={5}>
                        <Box flex={1}>
                            <Text fontWeight={'bold'}>{item.product.title}</Text>
                        </Box>
                        <Box w={{ base: '200px' }} textAlign={{base: 'left', md: 'right'}}>{item.price}</Box>
                        <Box w={{ base: '200px' }} textAlign={{base: 'left', md: 'right'}}>{item.quantity}</Box>
                        <Box w={{ base: '200px' }} textAlign={{base: 'left', md: 'right'}}>{(item.quantity * item.price).toFixed(2)}</Box>
                    </Flex>
                </Flex>
            ))}
            <Flex w={'100%'} bgColor={'gray.100'} p={5}>
                <Spacer></Spacer>
                <Center>
                    <Text fontSize='xl'>Order Total:</Text>
                    <Text fontSize='xl' ml={3} fontWeight={'bold'}>{getTotalAmount()}</Text>
                </Center>
            </Flex>
        </VStack>
    )
}

export default OrderDetailItems