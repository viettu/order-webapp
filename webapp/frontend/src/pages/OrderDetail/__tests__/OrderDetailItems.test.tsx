import React from 'react';
import OrderDetailItems from '../OrderDetailItems';
import renderer from 'react-test-renderer';
import { TestComponentWrapper } from '../../../utils';

test('snapshot', () => {
  const orderItems = [
    {
      id: 1,
      productId: 1,
      productTitle: 'AIR',
      productImage: 'air',
      price: 500,
      quantity: 5,
    },
  ];
  const component = renderer.create(
    <TestComponentWrapper>
      <OrderDetailItems items={orderItems}></OrderDetailItems>
    </TestComponentWrapper>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
