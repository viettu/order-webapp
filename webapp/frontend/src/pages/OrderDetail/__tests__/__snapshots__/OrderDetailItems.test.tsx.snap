// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
Array [
  <div
    className="chakra-stack css-1vr2anf"
  >
    <div
      className="css-1orwh6x"
    >
      <img
        alt="AIR"
        className="chakra-image css-b471h5"
        src="/images/air.png"
      />
      <div
        className="css-111yils"
      >
        <div
          className="css-1rr4qq7"
        >
          <p
            className="chakra-text css-722v25"
          >
            AIR
          </p>
        </div>
        <div
          className="css-1b81pbg"
        >
          $
          500
        </div>
        <div
          className="css-1b81pbg"
        >
          5
        </div>
        <div
          className="css-1b81pbg"
        >
          $
          2500
        </div>
      </div>
    </div>
    <div
      className="chakra-stack__divider css-1vk75av"
    />
    <div
      className="css-13jd5f3"
    >
      <div
        className="css-17xejub"
      />
      <div
        className="css-gmuwbf"
      >
        <p
          className="chakra-text css-mcpu91"
        >
          Order Total:
        </p>
        <p
          className="chakra-text css-37vpel"
        >
          $
          2500
        </p>
      </div>
    </div>
  </div>,
  <span />,
  <span
    className="chakra-env"
    hidden={true}
  />,
]
`;
